designs/SearchCommandSpec.md

```diff
@@ -0,0 +1,98 @@
+# Search CLI Tool for NuGet.exe and Dotnet.exe
+
+* Status: **Implementing**
+* Author(s): [Advay Tandon](https://github.com/advay26)
+* Issue: [9704](https://github.com/NuGet/Home/issues/9704) - Search CLI Tool for NuGet.exe and Dotnet.exe
+
+## Problem Background
+
+Search functionality from the command line for NuGet.exe and the .NET Core CLI has been a feature request from the NuGet open source community for some time. Additionally, the NuGet API is already equipped to handle search requests, so an extension of this service to command line users would be both practical to implement and also highly valuable.
+
+## Who are the customers
+
+All NuGet.exe and .NET CLI users.
+
+## Goals
+
+* Providing search functionality for NuGet packages through the commandline
+* Displaying the search results to users in a way that maximizes the visibility of the most relevant results.
+
+## Solution
+
+Th search feature leverages the search service provided by the NuGet API. Search targets multiple sources and displays all of their results one after the other. The command line interface is simple: __./NuGet.exe search \<query\>__.
+```
+PS C:\> ./NuGet.exe search logging
+```
+
+Each search result displays the name, version, number of downloads, and description of the package. Additionally, Search uses pagination to provide a clean viewing experience for users as they look at the results of their search. This ensures that users have access to the search results in order of decreasing relevance. A single page of results is shown at a time, and the user can look at the next page by pressing the __space__ key.
```

Comment:
Can sort be specified? eg, `-sort downloads`, or `-sort name`? 

What's the default sort? Should it be relevancy, download count, or name? 
I think some API's are Server dependent (either registration or search), so depending on which you use, it might be nice to tell the user what order they're about to see, and if they can change that.

Reply:
NuGet search protocol does not support sorting today. I'd suggest keeping this out of the MVP

Reply:
The only sort provided by our official API is relevance. /cc @t-mog-msft who is plumbing some of this through. downloads and name are not supported by API and there is no concrete plan for this yet.

Reply:
So the client won't be sorting, gotcha.
I'd still like to see some note to the user, like `(sorted by relevance according to the source)` ? 
We get so many questions about lack of sort/filter, so somehow stating it might be valuable. At least consider it for `-verbosity detailed`.